misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJob.kt

```diff
@@ -55,6 +58,36 @@ internal class SqsJob(
     metrics.jobsDeadLettered.labels(queueName.value, queueName.value).inc()
   }
 
+  /**
+   *  Assign a visibility timeout for some duration X making the job invisible to the consumers for
+   *  that duration. With every subsequent retry the duration becomes longer until it hits the max
+   *  value of 10hrs.
+   */
+   fun delayForFailure(maxRetryCount: Int = 10) {
+    val visibilityTime = calculateVisibilityTimeOut(maxRetryCount)
+    queue.call { client ->
+      client.changeMessageVisibility(
+        ChangeMessageVisibilityRequest()
+          .withQueueUrl(queue.url)
+          .withReceiptHandle(message.receiptHandle)
+          .withVisibilityTimeout(visibilityTime)
+      )
+    }
+    metrics.visibilityTime.labels(queueName.value, queueName.value).set(visibilityTime.toDouble())
+  }
+
+  private fun calculateVisibilityTimeOut(maxReceiveCount: Int): Int {
+    val currentReceiveCount = attributes["ApproximateReceiveCount"]?.toInt() ?: 1
+
+    val consecutiveRetryCount = (currentReceiveCount + 1).coerceAtMost(maxReceiveCount)
+    val backoff = 2.0.pow((consecutiveRetryCount - 1).toDouble()).toLong()
```

Comment:
Could we do `1L shl (consecutiveRetryCount - 1)`? Would avoid precision loss/casting and is much faster

Reply:
hmm not sure about it being much faster as i think JVM optimizes a lot under the hood. it does look cleaner but im personally not a big fan of bit operations as it hinders readability